mlir/lib/Dialect/Rock/utility/AmdArchDb.cpp

```diff
@@ -73,11 +75,36 @@ static constexpr AmdArchInfo
               /*hasFp8ConversionInstrs=*/false,
               /*hasOcpFp8ConversionInstrs=*/false, /*maxNumXCC=*/1);
 
+static AmdArchInfo fetchNativeArchInfo(unsigned long long deviceId = 0) {
+  hipDeviceProp_t prop;
+  if (auto err = hipGetDeviceProperties(&prop, deviceId); err != hipSuccess) {
+    llvm::errs() << "hipGetDeviceProperties error: " << hipGetErrorString(err)
+                 << "; Device ID: " << deviceId
+                 << "; Falling back to defaults\n";
+    return gcnInfo;
+  }
+
+  auto ret = lookupArchInfo(prop.gcnArchName); // get baseline
+  ret.waveSize = prop.warpSize;
```

Comment:
Can you please check what warpSize are you getting for the Navi ?

Reply:
On Navi4x, I get a warp size of 32.